junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertionsTests.java

```diff
@@ -2743,6 +2746,406 @@ void assertArrayEqualsDifferentNestedObjectArraysAndMessageSupplier() {
 		}
 	}
 
+	// --- assertIterableEquals --------------------------------------------
+
+	private static final class I implements Iterable<Object> {
+
+		private final List<Object> elements;
+
+		static I of(Object... elements) {
+			return new I(elements);
+		}
+
+		private I(Object... elements) {
+			this.elements = asList(elements);
+		}
+
+		@Override
+		public Iterator<Object> iterator() {
+			return elements.iterator();
+		}
+
+	}
+
+	@Test
+	void assertIterableEqualsObjectIterablesAndObjectLists() {
```

Comment:
Currently there are too many different logical assertions in this single unit test. This breaks the "only one assertion" rule. (see http://xp123.com/articles/3a-arrange-act-assert/).

Moreover, the combination of `asList` and `I.of` is very confusing.

Please think about applying the following renamings:

- Change name of private class `I` to `IterableFactory`
- Add factory method `IterableFactory.listOf(Object... elements)`
- Add factory method `IterableFactory.setOf(Object... elements)`
- Add factory method `IterableFactory.arrayOf(Object... elements)`

This also makes it more obvious, that the type of the `Iterable` is not taken into account.